source/manual/intro-to-docker.html.md

```diff
@@ -0,0 +1,331 @@
+
+```shell
+$mac docker build -t content-publisher
+```
+
+* Re-run docker with a new flag, `--privileged`, which gives the build user elevated permissions to run postgres:
+
+```shell
+$mac docker run -it --rm -v $(pwd):/app -v content-publisher-bundle:/usr/local/bundle --privileged content-publisher bash
+```
+
+#### Install Chrome for feature tests
+
+Test will still be failing because we have one have another dependency for Chrome.
+
+* Add the following to the Dockerfile:
+
+```docker
+RUN apt-get install -y libxss1 libappindicator1 libindicator7 && apt-get clean
+
+RUN wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
+
+RUN dpkg -i google-chrome-stable_current_amd64.deb; apt-get -fy install
+```
+
+* Then rebuild image and re-run the container:
+
+```shell
+$mac docker build -t content-publisher
+$mac docker run -it --rm -v $(pwd):/app -v content-publisher-bundle:/usr/local/bundle --privileged content-publisher bash
+```
+
+* Run the tests again:
+
+```shell
+$dev cd /apps
+$dev bundle exec rake
+```
+
+### Fixing IP with Docker Network
+
+IP addresses will be randomly assigned each time the container starts up, which can be a pain when you want containers to talk to each other. As was the case between the [content-publisher] and PostgreSQL containers we've been using so far.
+
+Docker has the notion of Networks which you can provide to each container as a connection that they can use.
+
+* Create a network:
+
+```shell
+$mac docker network create content-publisher-network
+```
+
+* Run the containers with the new network:
+
+```shell
+$mac docker run -it --rm -v content-publisher-postgres:/var/lib/postgresql/data --network content-publisher-network --network-alias postgres postgres
+
+$mac docker run -it --rm -v $PWD:/app -v content-publisher-bundle:/usr/local/bundle --privileged --network content-publisher-network -e TEST_DATABASE_URL=postgresql://postgres@postgres/content-publisher-test -e DATABASE_URL=postgresql://postgres@postgres/content-publisher-dev content-publisher:latest bash
+```
+
+### All in one line command
+
+Combining all of the above into a single line to execute:
+
+```
+docker run -it --rm -v $PWD:/app -v content-publisher-bundle:/usr/local/bundle --privileged --network content-publisher-network -e TEST_DATABASE_URL=postgresql://postgres@postgres/content-publisher-test -e DATABASE_URL=postgresql://postgres@postgres/content-publisher-dev -w /app content-publisher:latest bash
+```
+
+This is a lot to run in one command!
+
+Luckily, we can convert this long command into some YAML configuration which defines all the volumes and dependencies. We can then run that YAML file use [docker-compose](https://docs.docker.com/compose/), for example:
+
+`docker-compose run —rm content-publisher bundle exec rake`
+
+We can even run the above command as a CLI script to make it shorter and more reusable between our repos. This is what [govuk-docker][govuk-docker] does.
```

Comment:
Would be helpful to link to a simple compose file that demos this - probably not the actual ones in the repo, since they have all the extra interpolation/stack/volume stuff.